events/notifications.go

```diff
@@ -296,11 +288,9 @@ func addNotificationEvent(ctx api.Context, event api.Event) error {
 			CelEnv:       celEnv,
 		}
 
-		if valid, err := expressionRunner.Eval(ctx, n.Filter); err != nil || !valid {
-			// We consider an error in filter evaluation is a failed filter check.
-			// Mostly, the filter check returns an error if the variable isn't defined.
-			// Example: If the filter makes use of `check` variable but the event is for
-			// incident creation, then the expression evaluation returns an error.
+		if valid, err := expressionRunner.Eval(ctx, n.Filter); err != nil {
+			logs.IfError(db.UpdateNotificationError(id, err.Error()), "failed to update notification")
```

Comment:
I don't see how this prevents the filter from being run again, and will cause the row to be continually be touched?

We should have a StopNotification call here - 

I also think most of these logic should move to the notifications package - the events package should just be handing it off with the environment. 

Why do we have a pkgNotification.getNotificationIds and a pkgNotification.GetNotification(id) - Why not just combine it?

Reply:
> I don't see how this prevents the filter from being run again, and will cause the row to be continually be touched?
> We should have a StopNotification call here -

The expression is not run if the notification is in error status (line 281)

> Why do we have a pkgNotification.getNotificationIds and a pkgNotification.GetNotification(id) - Why not just combine it?

One's fetching all notifications that are listening on an event and the other one is fetching by given ids

> Also not sure why we have pkgNotification and pkgResponder, why not just use notification?

I think `responder` & `notification` are already used as variables so the package name was prefixed with `pkg`